pkg/pillar/zedcloud/lookupproxy.go

```diff
@@ -121,3 +121,56 @@ func LookupProxy(status *types.DeviceNetworkStatus, ifname string,
 	log.Infof("LookupProxy: No proxy configured for port %s", ifname)
 	return nil, nil
 }
+
+// IntfLookupProxyCfg - check if the intf has proxy configured
+func IntfLookupProxyCfg(status *types.DeviceNetworkStatus, ifname, downloadURL string) string {
+	var mayChange bool
+	// if proxy is not on the intf, then don't change anything
+	// if download URL has "http://" or "https://" then no change here regardless of proxy
+	// if there is proxy on this intf, treeat empty url scheme as for https,
+	// treat passed in download url schem docker://" and maybe sftp://just as https
+	if strings.Contains(downloadURL, "http") {
+		return downloadURL
+	} else if strings.Contains(downloadURL, "docker:") {
+		mayChange = true
+	}
+
+	var scheme string
+	for _, port := range status.Ports {
+		if ifname != port.IfName {
+			continue
+		}
+		var hasHTTP, hasHTTPS bool
+		for _, proxy := range port.ProxyConfig.Proxies {
+			switch proxy.Type {
+			case types.NPT_HTTP:
+				hasHTTP = true
+			case types.NPT_HTTPS:
+				hasHTTPS = true
+			}
+		}
```

Comment:
This might work when the proxies are explicitly configured, but if there instead is a pac file (whether specified in config or downloaded using a wpad URL), you'd need to execute that file to determine which type of proxy it would use.

Reply:
@gkodali-zededa can you take a look?

Reply:
Something similar already happens inside LookupProxy. Why is this new function required?

Reply:
@gkodali-zededa it still goes through the LookupProxy(), but here it tries to handle the case the 'download url' does not include http or https, or it has 'docker://', so this function is for format a 'new' downloadURL if there is proxy on the link

Reply:
@eriknordmark right, also need to handle the pac file cases. so instead of looping check on port.PorxyConfig.Proxies above, here it can 'privately' call the LookupProxy() inside this function to determine if the proxy is on this intf or not. Do it twice, one for 'http' and other for 'https'.

Reply:
But that means calling LookupProxy twice, which seems a bit inefficient. Can we do it only once?

Reply:
Right. But to change this is going to rework the LookupProxy(), probably not worth it. Since download is not a that frequent operation on the device.

Reply:
So for most cases we have http:// or https:// in the URLs? 
We'll do some more work for docker:// and when the scheme is unspecified. Does the unspecified scheme appear for VM image download from a datastore, or for something else?

And does this have any impact on the remote console? It uses different schemes, right?

Reply:
at least in our zedcloud UI point of view, the scheme completely depending on the user definition in the image FQDN, image path part of the string. Like in S3 datastore, it defines the "https://" in front of the path, but on Azure, it does not have anything.
No, this is purely downloading, nothing related to the remote console. Our code put "wss://" to start the remote console actions.